File: rentacar-frontend/src/services/carService.ts
import axiosInstance from './axiosInstance';
import { type CarResponse, type CreateCarRequest, type UpdateCarRequest, type PageResponse } from '../models/types';

const CarService = {
    getAll: async (): Promise<CarResponse[]> => {
        const response = await axiosInstance.get<PageResponse<CarResponse>>('/cars');
        return response.data.content;
    },

    add: async (request: CreateCarRequest): Promise<void> => {
        await axiosInstance.post('/cars', request);
    },

    update: async (request: UpdateCarRequest): Promise<void> => {
        await axiosInstance.put('/cars', request);
    },

    delete: async (id: number): Promise<void> => {
        await axiosInstance.delete(`/cars/${id}`);
    },

    uploadImage: async (carId: number, file: File): Promise<void> => {
        const formData = new FormData();
        formData.append('file', file);
        await axiosInstance.post(`/cars/${carId}/image`, formData, {
            headers: { 'Content-Type': 'multipart/form-data' }
        });
    }
};

export default CarService;

File: rentACar/src/main/java/com/sinan/rentACar/business/abstracts/CarService.java
package com.sinan.rentACar.business.abstracts;

import java.io.IOException;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import org.springframework.web.multipart.MultipartFile;

import com.sinan.rentACar.business.requests.CreateCarRequest;
import com.sinan.rentACar.business.requests.UpdateCarRequest;
import com.sinan.rentACar.business.responses.GetAllCarsResponse;

public interface CarService {
    Page<GetAllCarsResponse> getAll(Pageable pageable);
    void add(CreateCarRequest createCarRequest);
    void update(UpdateCarRequest updateCarRequest);
    void delete(int id);
    void uploadImage(int carId, MultipartFile file) throws IOException;
}
